src/EnergyPlus/OutputProcessor.cc

```diff
@@ -3453,169 +3214,100 @@ void UpdateDataandReport(EnergyPlusData &state, OutputProcessor::TimeStepType co
 
     // Main "Record Keeping" Loops for R and I variables
     for (auto *var : op->outVars) {
-        if (var->varType != VariableType::Real) continue;
         if (var->timeStepType != t_TimeStepTypeKey) continue;
 
-        OutVarReal *rVar = dynamic_cast<OutVarReal *>(var);
-        assert(rVar != nullptr);
+        Real64 value = (var->varType == VariableType::Real) ?
+            *(dynamic_cast<OutVarReal *>(var))->Which :
+            *(dynamic_cast<OutVarInt *>(var))->Which;
 
-        rVar->Stored = true;
-        if (rVar->storeType == StoreType::Averaged) {
-            Real64 CurVal = (*rVar->Which) * rxTime;
-            //        CALL SetMinMax(RVar%Which,MDHM,RVar%MaxValue,RVar%maxValueDate,RVar%MinValue,RVar%minValueDate)
-            if ((*rVar->Which) > rVar->MaxValue) {
-                rVar->MaxValue = (*rVar->Which);
-                rVar->maxValueDate = MDHM;
-            }
-            if ((*rVar->Which) < rVar->MinValue) {
-                rVar->MinValue = (*rVar->Which);
-                rVar->minValueDate = MDHM;
+        var->Stored = true;
+
+        if (var->storeType == StoreType::Average) {
+            Real64 CurVal = value * rxTime;
+            // TODO: Is this correct? Integer logic is different
```

Comment:
Here is another fun discrepancy between handing of int and real variables. Why is this logic done with the current `value` for real variables, but with `value * rxtime` for int variables? And why is this logic different only for `Average` variables and not for `Sum` variables?

Reply:
That's a great question. It's only different for averaged variables, and it's been this way since at least v2.2 (sadly, OutputProcessor.f90 is missing from the EnergyPlusArchive repo, but my local source code archive goes back to v2.2 but with no blame history). 

A quick test with 5ZoneAirCooled shows that the integer, real, and meter max/min values look correct.
```
idf (max/min are reported to eso only for daily or longer reporting frequency):
  Output:Variable,*,Site Day Type Index,monthly; !- Zone Average []
  Output:Meter:MeterFileOnly,Electricity:Facility,timestep;
  Output:Meter:MeterFileOnly,Electricity:Facility,monthly;

eso:
11,9,Environment,Site Day Type Index [] !Monthly [Value,Min,Day,Hour,Minute,Max,Day,Hour,Minute]
11,4.000000,1, 3, 1,15,7, 2, 1,15

455,7,WALL-1PF,Surface Inside Face Temperature [C] !Daily [Value,Min,Hour,Minute,Max,Hour,Minute]
455,17.068063288058173,12.94555448490067, 7,15,20.450554731139146,16,60
```
![image](https://github.com/NREL/EnergyPlus/assets/4991075/141d092f-015e-433d-8fd6-ee172f18702c)

```
mtr:
16,9,Electricity:Facility [J] !Monthly [Value,Min,Day,Hour,Minute,Max,Day,Hour,Minute]
16,12297200222.187939,766225.3673340459, 1, 1,15,11339356.444378309,17,15,15
```
![image](https://github.com/NREL/EnergyPlus/assets/4991075/0bd953a6-3dea-446a-a184-dc99c69fb375)

If `CurVal` is correct for integer vars, then why isn't is correct for real vars? It doesn't seem possible that both of these are producing correct min/max results, but they appear to.

Reply:
There are arbitrary inconsistencies like these all over EnergyPlus.